table/tables/tables.go

```diff
@@ -576,6 +576,21 @@ func TryGetCommonPkColumnIds(tbl *model.TableInfo) []int64 {
 	return pkColIds
 }
 
+// PrimaryPrefixColumnIDs get prefix column ids in primary key.
+func PrimaryPrefixColumnIDs(tbl *model.TableInfo) (prefixCols []int64) {
+	for _, idx := range tbl.Indices {
+		if !idx.Primary {
+			continue
+		}
+		for _, col := range idx.Columns {
+			if tbl.Columns[col.Offset].Flen > col.Length {
```

Comment:
I think `col.Length != types.UnspecifiedLength` should be enough because it is impossible to build a table that ` tbl.Columns[col.Offset].Flen < col.Length`
```sql
mysql> create table t (a char(1), primary key(a(2)));
ERROR 1089 (HY000): Incorrect prefix key; the used key part isn't a string, the used length is longer than the key part, or the storage engine doesn't support unique prefix keys
```